hastexo_guacamole_client/guacamole/client.py

```diff
@@ -0,0 +1,221 @@
+"""
+The underlying work for this code is published under the MIT License (MIT)
+(https://pypi.org/project/pyguacamole/)
+"""
+
+import socket
+import logging
+
+from .exceptions import GuacamoleError
+
+from .instruction import INST_TERM
+from .instruction import GuacamoleInstruction as Instruction
+
+# supported protocols
+PROTOCOLS = ('vnc', 'rdp', 'ssh')
+
+PROTOCOL_NAME = 'guacamole'
+
+BUF_LEN = 4096
+
+guac_logger = logging.getLogger(__name__)
+guac_logger.setLevel(logging.INFO)
+guac_logger.handlers = [logging.StreamHandler()]
+
+
+class GuacamoleClient(object):
+    """Guacamole Client class."""
+
+    def __init__(self, host, port, timeout=20, loglevel=logging.DEBUG, logger=None): # noqa E501
+        """
+        Guacamole Client class. This class can handle communication with guacd
+        server.
+
+        :param host: guacd server host.
+
+        :param port: guacd server port.
+
+        :param timeout: socket connection timeout.
+
+        :param debug: if True, default logger will switch to Debug level.
+        """
+        self.host = host
+        self.port = port
+        self.timeout = timeout
+
+        self._client = None
+
+        # handshake established?
+        self.connected = False
+
+        # Receiving buffer
+        self._buffer = bytearray()
+
+        # Client ID
+        self._id = None
+
+        self.logger = guac_logger
+        if logger:
+            self.logger = logger
```

Comment:
Hmm? Why not simply use `guac_logger` as the default of the `logger` argument; doesn't that achieve the same?

Reply:
Fixed in ea3cebaa11390a4b50b70df8cec8e09efc01283b.